### ISSUE TITLE
Feature request

### ISSUE THREAD

**Issue by netrunner-exe** [felixrosberg/FaceDancer#3]
Hello @felixrosberg!  I have a couple of ideas and suggestions to improve the user experience, and in general, if it could be implemented, it would be very convenient!

First request: is it possible to replace only one face in a video or photo, or only a specific selected face, in addition to replacing all defined faces?  It would be very convenient not only to replace all faces at once, but also the ability to choose which face.

The second request: is it possible to add the ability to use a face parsing mask, in addition to the current mask?  I found a couple of repositories, something like BisenetV2 on Tensorflow, that could be used for this purpose.  If this function were implemented, it would make it possible to make a more accurate mask along the contours of the face and also exclude from the mask those parts of the face that either turned out badly or that I would like to exclude from the process.  That is, we make a mask from the target face, blur around the edges and apply this mask to the result before returning to the image.  It would just be ideal to make an option like --parsing_mask True or False so that if such a result is needed, then use this function.  As practice shows, this solution has proven itself well and it would be really cool to see this implementation in FaceDancer!  Thank you!

Repos:
https://github.com/xuigo/Face_Parsing_BiseNetV2
https://github.com/nikhilroxtomar/Human-Face-Segmentation-in-TensorFlow
https://github.com/zllrunning/face-parsing.PyTorch

**Comment by felixrosberg** [comment]
Hello @netrunner-exe 

I do have a few scripts that have different functionalities that I will try clean up and push soon. I have been looking into a face segmentation blending option, I just need to find the time to implement and integrate it.

Regarding choosing face: This should be doable assuming the user know how many faces that is expected to be detected. But the user has to themselves figure out which detection is which face. I will look into this.

**Comment by ILG2021** [comment]
The user can draw a mask, so retinaface can detect the face only in the mask to ensure which face the user want to change. I also wana the feature. Very needed.

**Comment by felixrosberg** [comment]
I am currently working on a tracking function for a different application.

However this could be used for the user to select pairs they want to swap in two folders.

Then this tracker can be initialized with this pairs, so if it finds a match in the frame it will swap with the assigned face pair. It is soon finished, so implementing a script for this option should be doable rather quick.

**Comment by ILG2021** [comment]
> > The user can draw a mask, so retinaface can detect the face only in the mask to ensure which face the user want to change. I also wana the feature. Very needed.
> 
> FaceDancer is a CLI program, adding the ability to select a specific face (for example, mark it with a square on the frame) means writing a GUI for FaceDancer.  It would be great if someone who has free time and the ability to make a simple GUI using steamlit or tkinter contributed to the development of the project :)

There is a gui on huggerface
